neuralmonkey/dataset.py

```diff
@@ -58,26 +59,31 @@ def load_dataset_from_files(
         name = _get_name_from_paths(series_paths_and_readers)
 
     if lazy:
-        return LazyDataset(name, series_paths_and_readers, series_outputs,
-                           preprocessors)
-
-    series = {key: list(reader(paths))
-              for key, (paths, reader) in series_paths_and_readers.items()}
-
-    if preprocessors is not None:
-        for src_id, tgt_id, function in preprocessors:
-            if src_id == tgt_id:
-                raise Exception(
-                    "Attempt to rewrite series '{}'".format(src_id))
-            if src_id not in series:
-                raise Exception(
-                    ("The source series ({}) of the '{}' preprocessor "
-                     "is not defined in the dataset.").format(
-                         src_id, function.__name__))
-            series[tgt_id] = list(map(function, series[src_id]))
-
-    dataset = Dataset(name, series, series_outputs)
-    log("Dataset length: {}".format(len(dataset)))
+        dataset = LazyDataset(name, series_paths_and_readers, series_outputs,
+                              preprocessors)
+        # type: Dataset
+    else:
+        series = {key: list(reader(paths))
+                  for key, (paths, reader) in series_paths_and_readers.items()}
+
+        if preprocessors is not None:
+            for src_id, tgt_id, function in preprocessors:
+                if src_id == tgt_id:
+                    raise Exception(
+                        "Attempt to rewrite series '{}'".format(src_id))
+                if src_id not in series:
+                    raise Exception(
+                        ("The source series ({}) of the '{}' preprocessor "
+                         "is not defined in the dataset.").format(
+                             src_id, function.__name__))
+                series[tgt_id] = list(map(function, series[src_id]))
+
+        # pylint: disable=redefined-variable-type
+        dataset = Dataset(name, series, series_outputs)
+        # pylint: enable=redefined-variable-type
+        log("Dataset length: {}".format(len(dataset)))
```

Comment:
beru tady ten if a ty řádky 81 - 84 jako nezměněný, jen šouplý doprava - je to tak?